activerecord/test/cases/adapters/postgresql/quoting_test.rb

```diff
@@ -18,6 +18,18 @@ def test_type_cast_false
           assert_equal false, @conn.type_cast(false)
         end
 
+        def test_type_cast_duration_with_cast_duration_enabled
+          with_cast_duration(true) do
+            assert_equal "PT1M", @conn.type_cast(1.minute)
+          end
+        end
+
+        def test_type_cast_duration_with_cast_duration_disabled
+          with_cast_duration(false) do
+            assert_raises(TypeError) { @conn.type_cast(1.minute) }
```

Comment:
The abstract adapter `type_cast` does not handle duration, so code that uses `type_cast` and not `quote` will raise if `cast_duration` is disabled, e.g. 
```rb
ActiveRecord::Base.connection.exec_query("select $1", nil, [1.minute])
```
but since this is the existing behavior, I didn't make an attempt to change it if the new setting is disabled.